tests/unit/lms/services/grouping_test.py

```diff
@@ -29,42 +29,28 @@ def test_upsert_with_parents_inserts(self, svc, db_session):
 
         assert db_session.query(CanvasGroup).one() == test_grouping
 
-    def test_upsert_with_parents_updates(self, svc, db_session):
+    def test_upsert_with_parent_updates_existing_groupings(self, svc, db_session):
         course = factories.Course()
-        tool_consumer_instance_guid = (
-            course.application_instance.tool_consumer_instance_guid
-        )
-        lms_id = "lms_id"
+        kwargs = {
+            "tool_consumer_instance_guid": course.application_instance.tool_consumer_instance_guid,
+            "lms_id": "lms_id",
+            "parent": course,
+            "type_": Grouping.Type.CANVAS_GROUP,
+        }
         old_name = "old_name"
         old_extra = {"extra": "old"}
         new_name = "new_name"
         new_extra = {"extra": "new"}
+        # Insert an existing grouping into the DB.
+        svc.upsert_with_parent(lms_name=old_name, extra=old_extra, **kwargs)
```

Comment:
I think the reason why this test has to call `upsert_with_parent()` as part of its arrange block (rather than just inserting the grouping into the DB itself) is that groupings are difficult to construct because you have to generate the `authority_provided_id`. You can't just do something like `db_session.add(Grouping(lms_name=old_name, extra=old_extra, **kwargs))`, it ends up being much more awkward than that. It might be possible to fix this with some refactoring.